### PR TITLE
Restructure proxy installation

### PR DESCRIPTION
**Pull request by piyush-garg**
Currently proxy webhook is installed in operator namespace as part of
operator. this refactoring will make it to install as a component like
pipeline and triggers in the tekton namespace and installation will be
done along with pipeline component.

This fixes the issue getting faced as mutatuingwebhookcnfiguration is not
supported in CSV if name is specified.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```release-note
Restructure proxy installation as a component
```

### CODE REVIEW

File: pkg/reconciler/kubernetes/tektonpipeline/tektonpipeline.go [hunk 1]
```diff
@@ -105,7 +106,7 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, tp *v1alpha1.TektonPipel
 		return err
 	}
 	stages := common.Stages{
-		common.AppendTarget,
+		r.appendPipelineTarget,
```
Comment (nikhil-thomas) [inline pkg/reconciler/kubernetes/tektonpipeline/tektonpipeline.go:109]: shall we keep these 2 as separate stages. ie, leave common.AppendTarget as it is. Then make r.appendPipelineTarget next stage and make it handle only what `addProxy` is doing.

In addition, after separating we can also rename the `r.appendPipelineTarget` to something like `r.appendProxy`

**Review by savitaashture — COMMENTED** [review]
/lgtm

File: pkg/reconciler/kubernetes/tektonpipeline/tektonpipeline.go (hunk 1 above)
Comment (piyush-garg) [reply pkg/reconciler/kubernetes/tektonpipeline/tektonpipeline.go:109]: I will work on this in a follow-up PR. Thanks

**Pull request merged**